Fix additional Clang unused warnings

donny-dont opened the pull request:
#### d3a6554b173379c0d3ab47f3b5dab7169c977f4e
<pre>
Fix additional Clang unused warnings
<a href="https://bugs.webkit.org/show_bug.cgi?id=246366">https://bugs.webkit.org/show_bug.cgi?id=246366</a>

Reviewed by Darin Adler.

Fix some code warnings from Clang 15 around unused values/code. Fixes
instances of `-Wunused-variable`, `-Wunused-lambda-capture`,
`-Wunused-but-set-variable`, and `-Wunused-function`.

* Source/JavaScriptCore/heap/GCSegmentedArrayInlines.h:
* Source/WebCore/platform/graphics/playstation/SystemFontDatabasePlayStation.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:

Canonical link: <a href="https://commits.webkit.org/255450@main">https://commits.webkit.org/255450@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b6a63ae9c955c2ba9689365daa7624aecf8175a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92557 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23144 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102308 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96558 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1774 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30143 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84968 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98470 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98219 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1193 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79056 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/28136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/83112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82812 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/71226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/83931 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36553 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/78969 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34335 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/17922 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/27374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3778 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38204 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/40529 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/81588 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40113 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37079 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18452 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/heap/GCSegmentedArrayInlines.h:
```diff
@@ -180,7 +180,7 @@ inline void GCSegmentedArray<T>::validatePrevious()
     for (GCArraySegment<T>* current = m_segments.head(); current; current = current->next())
         count++;
-    ASSERT(m_segments.size() == m_numberOfSegments);
+    ASSERT_UNUSED(count, m_segments.size() == m_numberOfSegments);
```
This makes no sense. Why are we computing `count` and not using it? I think the code above to compute count is dead code. If we are computing `count` then it seems we should be asserting it is `== m_segments.size()`?

donny-dont replied:
I'm assuming there's some kind of validation happening here but maybe someone on the @WebKit/jsc-reviewers side can verify

Constellation replied:
I think the old ASSERT looks wrong. We should do `ASSERT(count == m_numberOfSegments)` probably here.

Constellation replied:
Or, maybe, we should skip count. `m_segments.size()` would be doing the same thing (let's check that).

donny-dont replied:
updating to `count == m_numberOfSegments`

darinadler replied:
> Or, maybe, we should skip count. `m_segments.size()` would be doing the same thing (let's check that).

Yes that’s what I was thinking too. Someone had the idea of using `m_segments.size()` but forgot to delete the code that does the same thing.

darinadler commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -485,7 +485,9 @@ void WebProcess::initializeWebProcess(WebProcessCreationParameters&& parameters)
 
     PAL::registerNotifyCallback("com.apple.WebKit.logPageState"_s, [this] {
         for (auto& page : m_pageMap.values()) {
+#if !RELEASE_LOG_DISABLED
```
Let’s put this `#if` around the entire `PAL::registerNotifyCallback` call and the entire lambda. There’s no need to do any of this if `RELEASE_LOG` is disabled.

mcatanzaro commented on Source/WebCore/platform/VideoPixelFormat.cpp:
```diff
@@ -43,6 +43,8 @@ VideoPixelFormat convertVideoFramePixelFormat(uint32_t format)
         return VideoPixelFormat::RGBA;
     ASSERT_NOT_REACHED();
+#else
+    UNUSED_PARAM(format);
```
I fixed this in https://github.com/WebKit/WebKit/pull/5202 but your change is byte-identical to mine, so instead of a conflict it should hopefully just disappear when you rebase.

donny-dont replied:
Had to update the commit message as well to reflect it.

donny-dont committed:
Fix additional Clang unused warnings
https://bugs.webkit.org/show_bug.cgi?id=246366

Reviewed by Darin Adler.

Fix some code warnings from Clang 15 around unused values/code. Fixes
instances of `-Wunused-variable`, `-Wunused-lambda-capture`,
`-Wunused-but-set-variable`, and `-Wunused-function`.

* Source/JavaScriptCore/heap/GCSegmentedArrayInlines.h:
* Source/WebCore/platform/graphics/playstation/SystemFontDatabasePlayStation.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:

Canonical link: https://commits.webkit.org/255450@main
